OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -48,6 +48,17 @@ public override Activity Tick(Actor self)
 
 			if (IsCanceling || target.Type == TargetType.Invalid)
 			{
+				// Make aircraft climb back to CruiseAltitude before moving on to NextActivity
+				if (NextActivity != null)
```

Comment:
I think we can skip this check and always do the height check / TakeOff?
Better to always do it here than to rely on two different cases (`Aircraft.OnBecomingIdle`) for idle/not idle handling.

We'll still need the code in Aircraft to handle edge cases like the terrain height changes during flight, but that is fine.

Reply:
Note that in that case this should be aware of `Aircraft.LandWhenIdle` as well.

Reply:
> Note that in that case this should be aware of Aircraft.LandWhenIdle as well.

Generally speaking I agree, but I'd like to leave migrating the current `OnBecomingIdle` cases to a dedicated follow-up, as my first attempt to cover this here promptly resulted in some nasty "activity went into a weird uncancellable loop".

This PR already has scope-creeped beyond what I originally wanted it to be, I'd like to leave further improvements to dedicated, purpose-focused follow-ups.